nucliadb_node/nucliadb_node/pull.py

```diff
@@ -61,369 +51,65 @@
-                    "storage_key": pb.storage_key,
-                    "time": time.time() - start,
-                },
-            )
-
-    async def subscribe(self):
-        logger.info(f"Last seqid {self.last_seqid}")
-        try:
-            await self.js.stream_info(const.Streams.INDEX.name)
+            await self.nats_connection_manager.js.stream_info(const.Streams.INDEX.name)
         except StreamNotFoundError:
             logger.info("Creating stream")
-            res = await self.js.add_stream(
+            await self.nats_connection_manager.js.add_stream(
                 name=const.Streams.INDEX.name,
                 subjects=[
                     const.Streams.INDEX.subject.format(node=">"),
                 ],
             )
-            await self.js.stream_info(const.Streams.INDEX.name)
+            await self.nats_connection_manager.js.stream_info(const.Streams.INDEX.name)
 
         subject = const.Streams.INDEX.subject.format(node=self.node)
-        res = await self.js.subscribe(
+        await self.nats_connection_manager.subscribe(
             subject=subject,
             queue=const.Streams.INDEX.group.format(node=self.node),
             stream=const.Streams.INDEX.name,
             flow_control=True,
+            manual_ack=True,
             cb=self.subscription_worker,
+            subscription_lost_cb=self.setup_nats_subscription,
             config=nats.js.api.ConsumerConfig(
-                deliver_policy=nats.js.api.DeliverPolicy.BY_START_SEQUENCE,
-                opt_start_seq=self.last_seqid or 1,
+                deliver_policy=nats.js.api.DeliverPolicy.ALL,
                 ack_policy=nats.js.api.AckPolicy.EXPLICIT,
                 max_ack_pending=nats_consumer_settings.nats_max_ack_pending,
                 max_deliver=nats_consumer_settings.nats_max_deliver,
                 ack_wait=nats_consumer_settings.nats_ack_wait,
                 idle_heartbeat=nats_consumer_settings.nats_idle_heartbeat,
             ),
         )
-        self.subscriptions.append(res)
         logger.info(f"Subscribed to {subject} on stream {const.Streams.INDEX.name}")
 
-
-class IndexedPublisher:
-    def __init__(self):
-        self.pubsub = None
-
-    async def initialize(self):
-        self.pubsub = await get_pubsub()
-
-    async def finalize(self):
-        await self.pubsub.finalize()
-
-    async def indexed(self, indexpb: IndexMessage):
-        if not indexpb.HasField("partition"):
-            logger.warning(f"Could not publish message without partition")
-            return
-
-        message = Notification(
-            partition=int(indexpb.partition),
-            seqid=indexpb.txid,
-            uuid=indexpb.resource,
-            kbid=indexpb.kbid,
-            action=Notification.INDEXED,
-        )
-
-        await self.pubsub.publish(
-            const.PubSubChannels.RESOURCE_NOTIFY.format(kbid=indexpb.kbid),
-            message.SerializeToString(),
-        )
+    @subscriber_observer.wrap()
```

Comment:
this metrics observer does not make sense anymore here, right? We should have observers at the workers/processors now

Reply:
I dont think its necessary. we have this metric already from the nucliadb_telemetry package